tests/testthat/test-derive_vars_crit_flag.R

```diff
@@ -0,0 +1,108 @@
+## Test 1: works with defaults ----
+test_that("derive_vars_crit_flag Test 1: works with defaults", {
+  expected <- tibble::tribble(
+    ~AVAL, ~CRIT1FL,      ~CRIT1,
+    23,    NA_character_, NA_character_,
+    42,    "Y",           "AVAL > 40",
+    NA,    NA_character_, NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = derive_vars_crit_flag(
+      select(expected, AVAL),
+      condition = AVAL > 40,
+      description = "AVAL > 40"
+    ),
+    keys = "AVAL"
+  )
+})
+
+## Test 2: create numeric flag ----
+test_that("derive_vars_crit_flag Test 2: create numeric flag", {
+  expected <- tibble::tribble(
+    ~AVAL, ~CRIT1FL,      ~CRIT1FLN, ~CRIT1,
+    23,    NA_character_, NA_real_,  NA_character_,
+    42,    "Y",           1,         "AVAL > 40",
+    NA,    NA_character_, NA_real_,  NA_character_
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = derive_vars_crit_flag(
+      select(expected, AVAL),
+      condition = AVAL > 40,
+      description = "AVAL > 40",
+      create_numeric_flag = TRUE
+    ),
+    keys = "AVAL"
+  )
+})
+
+## Test 3: using values Y and N ----
+test_that("derive_vars_crit_flag Test 3: using values Y and N", {
+  expected <- tibble::tribble(
+    ~PARAMCD, ~AVAL, ~CRIT2FL,      ~CRIT2FLN, ~CRIT2,
+    "AST",    23,    "N",           0,         "AST > 40",
+    "AST",    42,    "Y",           1,         "AST > 40",
+    "AST",    NA,    NA_character_, NA_real_,  "AST > 40",
+    "ALT",    26,    "N",           0,         "ALT > 40",
+    "ALT",    56,    "Y",           1,         "ALT > 40",
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = derive_vars_crit_flag(
+      select(expected, PARAMCD, AVAL),
+      crit_nr = 2,
+      condition = AVAL > 40,
+      description = paste(PARAMCD, "> 40"),
+      values_yn = TRUE,
+      create_numeric_flag = TRUE
+    ),
+    keys = "AVAL"
+  )
+})
+
+## Test 4: error if invalid condition ----
+test_that("derive_vars_crit_flag Test 4: error if invalid condition", {
```

Comment:
Can we specify that the variable is missing in the input dataset

Reply:
Done